.buildkite/pipeline_ab.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Generate Buildkite performance pipelines dynamically"""
+import os
+
+from common import COMMON_PARSER, group, overlay_dict, pipeline_to_json
+
+perf_test = {
+    "snapshot-latency": {
+        "label": "📸 Snapshot Latency",
+        "test_path": "integration_tests/performance/test_snapshot_ab.py",
+        "devtool_opts": "-c 1-12 -m 0",
+        "timeout_in_minutes": 60,
+    },
+}
+
+
+def build_group(test):
+    """Build a Buildkite pipeline `group` step"""
+    devtool_opts = test.pop("devtool_opts")
+    test_path = test.pop("test_path")
+    revision_a = os.environ["REVISION_A"]
+    revision_b = os.environ["REVISION_B"]
+    return group(
+        label=test.pop("label"),
+        command=f"./tools/devtool -y test --ab {devtool_opts} -- {revision_a} {revision_b} --test {test_path}",
+        artifacts=["./test_results/*"],
+        instances=test.pop("instances"),
+        platforms=test.pop("platforms"),
+        # and the rest can be command arguments
+        **test,
+    )
+
+
+parser = COMMON_PARSER
```

Comment:
Nit
```suggestion
parser = copy.deepcopy(COMMON_PARSER)
```
requires
```rust
import copy
```

Reply:
Mh, not really sure why we would need a deepcopy of the argparse.ArgumentParser, I can't think of any scenario where we would want two different argument parsers in the same script.

Reply:
Yeah it should be fine. Just the rustacean in me alerted at what is effectively `static mut COMMON_PARSER`.